File: LeetCode/3484/golang/lc3484_test.go
package lc3484

import "testing"

func Test_Spreadsheet(t *testing.T) {
	s := Constructor(3)
	if actual := s.GetValue("=5+7"); actual != 12 {
		t.Errorf("Operation =5+7 failed: want 12, got %d", actual)
	}

	s.SetCell("A1", 10)
	if actual := s.GetValue("=A1+6"); actual != 16 {
		t.Errorf("Operation =A1+6 failed: want 16, got %d", actual)
	}

	s.SetCell("B2", 15)
	if actual := s.GetValue("=A1+B2"); actual != 25 {
		t.Errorf("Operation =A1+B2 failed: want 25, got %d", actual)
	}

	s.ResetCell("A1")
	if actual := s.GetValue("=A1+B2"); actual != 15 {
		t.Errorf("Operation =A1+B2 after resetting A1 failed: want 15, got %d", actual)
	}

	// Test whether not-yet-initialized values are returned properly as zero:
	if actual := s.GetValue("=C1+12345"); actual != 12345 {
		t.Errorf("Operation =C1+12345 failed: want 12345, got %d", actual)
	}
}

File: LeetCode/3484/python/lc3484_test.py
import unittest

from lc3484 import Spreadsheet

class Test_spreadsheet(unittest.TestCase):
    def test(self):
        s = Spreadsheet(3)
        self.assertEqual(s.getValue("=5+7"), 12)

        s.setCell("A1", 10)
        self.assertEqual(s.getValue("=A1+6"), 16)
        self.assertEqual(s.getValue("=6+A1"), 16)

        s.setCell("B2", 15)
        self.assertEqual(s.getValue("=A1+B2"), 25)

        s.resetCell("A1") # same as setCell("A1", 0). A1 is now 0.
        self.assertEqual(s.getValue("=A1+B2"), 15)


        s = Spreadsheet(458)
        self.assertEqual(s.getValue("=O126+10272"), 10272)

        s = Spreadsheet(24)
        s.setCell("B24", 66688)
        s.resetCell("O15")
        self.assertEqual(s.getValue("=B24+O15"), 66688)

if __name__ == "__main__":
    unittest.main()
